### ISSUE TITLE
[fvt]xcatprobe image will loop through all node even  I input only one node

### ISSUE THREAD

**Issue by junxiawang** [xcat2/xcat-core#1417]
env: all linux platform
build:root@c910f03c05k05:/opt/xcat/bin# lsxcatd  -v
Version 2.12.1 (git commit 5431a2ab9ccd42f5f76540ec0868429dd6199056, built Sun Jun 26 06:47:12 EDT 2016)
How to reproduce:
```
root@c910f03c05k05:/opt/xcat/bin# ./xcatprobe image -c c910f03c05k09
Smartmatch is experimental at ./xcatprobe line 64.
Smartmatch is experimental at ./xcatprobe line 173.
node is c910f03c05k02
       c910f03c05k02 is diskless
node is c910f03c05k09
       c910f03c05k09 is diskless
[ OK ] Pinging c910f03c05k02
[ OK ] Pinging c910f03c05k09
       ---- Gathering information from node c910f03c05k02 ----
       ---- Gathering information from node c910f03c05k09 ----
[FAIL] Not all compute nodes are installed with the same os image.
c910f03c05k02   rhels7.2-ppc64le-netboot-compute : fd7eeb7e-f243-4d4c-9884-d3409c8f8668
c910f03c05k09   ubuntu14.04.4-ppc64el-netboot-compute : 81cd45bf-e463-42ee-bc21-e86e21ddeec4

root@c910f03c05k05:/opt/xcat/bin# lsdef -v
lsdef - Version 2.12.1 (git commit 5431a2ab9ccd42f5f76540ec0868429dd6199056, built Sun Jun 26 06:47:12 EDT 2016)
root@c910f03c05k05:/opt/xcat/bin# 
```
------------------------------> I input only one node 

```
my @nodes = `nodels`;
@nodes = grep(s/^[\s]+|[\s]*$//g, @nodes);

my @pingable_nodes;
my @diskless_nodes;
my $na = "N/A";

# First, extract diskless nodes
foreach (@nodes) {
    my $lsdef_provmethod = `lsdef $_ -i provmethod -c`;
    print "node is $_\n";
```
--------------------> from code get that it loops through all nodes

**Comment by daniceexi** [comment]
@junxiawang Please update the format of your messages.

**Comment by daniceexi** [comment]
@gurevichmark I think this is a problem. ``image`` subcommand should accept the noderange instead of always trying to compare against all the nodes. In a large cluster with thousands of nodes, it's not possible to have all the nodes with same osimage. Mostly they have multiple groups, each group uses same osimage.

**Comment by gurevichmark** [comment]
I would like to propose this as a future enhancement. Currently the image probe will be useful to the Coral team where all the nodes in the cluster need to be checked to be the same. Perhaps in the next release `image` probe can be improved to take a `noderange` and make it useful to a more general xCAT user.

**Comment by whowutwut** [comment]
@daniceexi @hu-weihua Does the `xcatprobe` command have libraries that will accept node range? Do we have that built into the framework so that the various probe commands can reuse that code?

**Comment by daniceexi** [comment]
Good idea, but current there's no common subroutine for the noderange handling.

**Comment by gurevichmark** [comment]
Fixed by pull request #1518

**Comment by gurevichmark** [comment]
@junxiawang Can you please verify this issue is resolved now.
It was fixed by pull request #1518
